### PR TITLE
Add `WorkQueue.status` to server and client side schemas

### LINKED ISSUE #11830: Add the `status` field to `WorkQueue` objects in OSS

### PR DESCRIPTION
**Pull request by urimandujano**
This PR adds the `status` field to `WorkQueue` objects and responses. The status is determined by the server when it receives a "poll" asking for a `WorkQueue` object and is conditional on the recency of the work queue's `last_polled` field. 

Closes #11830 

### Example
Work Queue responses now include a status field, determined via `last_polled` and `is_paused`:
```json
{
    "id": "6648ad79-a05a-490c-ac58-0f4eea8461ee",
    "created": "2024-02-02T21:58:28.664611+00:00",
    "updated": "2024-02-02T21:58:28.834000+00:00",
    "name": "wq-1",
    "description": "All about my work queue",
    "is_paused": true,
    "concurrency_limit": null,
    "priority": 1,
    "work_pool_id": "ad6678dd-e13e-4a84-92ba-cb851658d7f4",
    "filter": null,
    "last_polled": null,
    "work_pool_name": "default-agent-pool",
    "status": "PAUSED"
}
```

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Comment by urimandujano** [comment]
I had a chance to verify this against the UI work we've done and the results look good:
image

File: src/prefect/server/schemas/responses.py [hunk 1]
```diff
@@ -303,13 +303,26 @@ class WorkQueueResponse(schemas.core.WorkQueue):
...
+        if response.is_paused:
+            response.status = schemas.statuses.WorkQueueStatus.PAUSED
+        else:
+            unready_at = datetime.datetime.now(
+                tz=datetime.timezone.utc
+            ) - datetime.timedelta(seconds=WORK_QUEUE_LAST_POLLED_TIMEOUT_SECONDS)
+            if response.last_polled and response.last_polled > unready_at:
+                response.status = schemas.statuses.WorkQueueStatus.READY
+            else:
+                response.status = schemas.statuses.WorkQueueStatus.NOT_READY
```
Comment (collincchoy) [inline src/prefect/server/schemas/responses.py:325]: might be nice to have a test case covering the `last_polled <= unready_at` => NOT_READY

File: src/prefect/server/schemas/responses.py [hunk 2]
```diff
@@ -303,13 +303,26 @@ class WorkQueueResponse(schemas.core.WorkQueue):
         default=None,
         description="The name of the work pool the work pool resides within.",
     )
+    status: Optional[schemas.statuses.WorkQueueStatus] = Field(
+        default=None, description="The queue status."
+    )
```
Comment (serinamarie) [inline src/prefect/server/schemas/responses.py:308]: nit: "The status of a work queue based on when it was last polled"

File: tests/client/test_prefect_client.py
Comment (serinamarie) [inline tests/client/test_prefect_client.py:1]: nice!

File: tests/server/api/test_work_queues.py [hunk 3]
```diff
@@ -495,6 +502,60 @@ async def test_read_work_queue_runs_associated_deployments_return_status_of_read
...
+            f"/work_queues/{work_queue.id}/get_runs",
+            json=dict(agent_id=str(uuid.uuid4())),
+        )
+        assert response.status_code == status.HTTP_200_OK
+
+        # Verify the work queue status is still PAUSED
+        wq_response = await client.get(f"/work_queues/{work_queue.id}")
+        assert wq_response.status_code == status.HTTP_200_OK
+        assert wq_response.json()["status"] == "PAUSED"
+
```
Comment (serinamarie) [inline tests/server/api/test_work_queues.py:558]: awesome "regression test" kinda

File: src/prefect/server/schemas/responses.py (hunk 1 above)
Comment (serinamarie) [reply src/prefect/server/schemas/responses.py:325]: I concur

**Review by serinamarie — COMMENTED** [review]
Looking good!

File: tests/server/api/test_work_queues.py (hunk 3 above)
Comment (urimandujano) [reply tests/server/api/test_work_queues.py:577]: If only we'd had this test on our other PR :)

File: src/prefect/server/schemas/responses.py (hunk 1 above)
Comment (urimandujano) [reply src/prefect/server/schemas/responses.py:325]: I've added a test that verifies that conditional branch!

**Review by collincchoy — APPROVED** [review]
🚀

**Pull request merged**